rcsb/db/wf/RepoLoadWorkflow.py

```diff
@@ -351,6 +408,106 @@ def splitIdList(self, op, **kwargs):
 
         return ok
 
+    def formatIdFileSource(self, idVal, source, noBcifSubdirs):
+        if noBcifSubdirs:
+            path = idVal + ".bcif"
+        else:
+            path = os.path.join(idVal[1:3], idVal + ".bcif")
+        return f"{idVal} {path} {source}"
+    
+    def pdbTimeStampCheck(self, hD, targetFileDir, targetFileSuffix):
+        res = hD.copy()
+        for id, value in hD.items():
+            pathToItem = os.path.join(targetFileDir, id + targetFileSuffix)
+            if Path(pathToItem).exists():
+                t1 = Path(pathToItem).stat().st_mtime
+                t2 = datetime.datetime.strptime(value, "%Y-%m-%dT%H:%M:%S%z").timestamp()
+                if t1 > t2:
+                    res.pop(id)
+        return res
+
+    def csmTimeStampCheck(self, hD, targetFileDir, targetFileSuffix):
+        res = hD.copy()
+        for id, value in hD.items():
+            value["modelPath"].lower()
+            pathToItem = os.path.join(targetFileDir, id + targetFileSuffix)
+            if Path(pathToItem).exists():
+                t1 = Path(pathToItem).stat().st_mtime
+                t2 = datetime.datetime.strptime(value["lastModifiedDate"], "%Y-%m-%dT%H:%M:%S%z").timestamp()
+                if t1 > t2:
+                    res.pop(id)
+        return res
```

Comment:
These 2 functions are almost identical. Can you not extract the difference to a parameter and then use only one function?